src/trace-api.ts

```diff
@@ -199,20 +199,6 @@ export class TraceAgent implements TraceAgentInterface {
 
     const rootSpan = cls.get().getContext();
     if (rootSpan.type === SpanDataType.ROOT) {
-      if (!!rootSpan.span.endTime) {
-        // A closed root span suggests that we either have context confusion or
-        // some work is being done after the root request has been completed.
-        // The first case could lead to a memory leak, if somehow all spans end
-        // up getting misattributed to the same root span – we get a root span
-        // with continuously growing number of child spans. The second case
-        // seems to have some value, but isn't representable. The user probably
-        // needs a custom outer span that encompasses the entirety of work.
-        this.logger!.warn(`TraceApi#createChildSpan: [${
-            this.pluginName}] Creating phantom child span [${
-            options.name}] because root span [${
-            rootSpan.span.name}] was already closed.`);
-        return UNCORRELATED_SPAN;
-      }
```

Comment:
This is a useful error message. I don't see how we can remove it without an equivalent error message some where else.

We still need to be able to distinguish context confusion cases from context loss cases. As an example, this error message is very useful in identifying situations with a continuation is correlated to the wrong RootSpan – this shows up in the background log flusing use-case for example. If a Log was initialized in the first context, and flushes in the background chained on some promise, it would appear to be a continuation of the first context. This error message catches that scenario.

Reply:
Added it back in. I think having this logging message available actually isn't incompatible with allowing sequential root spans as I had originally thought.